### DISCUSSION TITLE
Putting a clip back on to the system clipboard

### DISCUSSION THREAD

**Discussion by KenJean** [Slackadays/Clipboard#108] [Q&A] [answered]
So I've set-up quite a few persistent clips in storage. At the terminal, I can easily access these with the `cb paste_<clip>` command. This much I understand.
When I wish to put a clip back on to the system clipboard in order to be able to paste it in, say, vim or firefox, I execute a `cb load_clip 0`
cb now shows that I have, indeed, placed the clip back on to clipboard0 (system clipboard.)
But when I do a paste in firefox, what gets pasted is the URL (i.e. the location) of the clip rather than the actual contents of the clip.
So I'm obviously missing something. How do return the contents of a clip back to where I can just Ctrl-V it into an application?

Otherwise, this thing works a treat. I love it!!

⇒ KJ

**Comment by Slackadays** [comment, marked as answer]
@KenJean Just fixed this in https://github.com/Slackadays/Clipboard/commit/6e924e5d6fcbc9c9de4065eeee939cfc7d2920d2 :smiley:

**Reply by KenJean** [reply]
You da man !!!!!
This must be some sort of land-speed record for bug-fixing.

I was using CopyQ until now, but I'm hoping to be able to dump that completely in favour of your wonderful programme.

⇒ KJ

**Comment by KenJean** [comment]
Oh dear,
Me thinks that while this fixed my problem, it caused the opposite action no longer to work. In other words, when I now copy something TO the system clipboard, it is no longer detected by cb.
Hope you can fix this.

⇒ KJ

**Reply by Slackadays** [reply]
What are the exact commands you're using?

**Comment by KenJean** [comment]
In this particular case, I highlight what I want copied in my browser, then I right click to copy it to the system clipboard. When I open an editor, I then Ctrl-V to paste what I just copied. This works fine.
This used to also register in cb as something that is now on clipboard0, now, it no longer does.

⇒ KJ

**Reply by Slackadays** [reply]
What system are you using and where did you upgrade Clipboard from? I can't find anything that might be causing your issue, as your current scenario works fine for me, so it could be an idiosyncrasy from compiling it yourself. For example, according to the Install Manually section in the readme, you'll need libx11 or libwayland plus wayland-protocols to get it to work with X11 or Wayland. And if it can't connect to X11/Wayland, then the system clipboard won't show up.

**Comment by KenJean** [comment]
OK, my bad. I just figured what I did what not kosher.
When you fixed it the first time, I just git cloned your repository, compiled for the binary and just used to binary directly rather than install it properly.
Sorry to cause you to go through this needless hassle.

⇒ KJ

**Reply by Slackadays** [reply]
That's ok, I'm always here to help!  :)
